src/hooks/useOrderData.ts

```diff
@@ -0,0 +1,19 @@
+import store from '../store';
+import { setData } from '../store/data';
+import getOrderDataRequest from '../utils/getOrderDataRequest';
+
+const getOrderData = async (id: string | undefined) => {
+  setInterval(async () => {
+    const dispatch = store.dispatch;
+    if (id) {
+      try {
+        const { data } = await getOrderDataRequest(id);
```

Comment:
if data is marked as `status: completed`, stop the interval until the id changes (using useEffect for example)